examples/src/main/scala/org/apache/spark/examples/ml/DataFrameExample.scala

```diff
@@ -54,14 +54,13 @@ object DataFrameExample {
       }
     }
 
-    parser.parse(args, defaultParams).map { params =>
-      run(params)
-    }.getOrElse {
-      sys.exit(1)
+    parser.parse(args, defaultParams) match {
+      case Some(params) => run(params)
+      case _ => sys.exit(1)
     }
```

Comment:
There is nothing incorrect about generating an expression of type Unit; what I am talking about is only pattern matching on Options, not the `view` issue; IJ has lots of opinions, not all of them correct.

What theoretical problem are you talking about with `fold` over `Option`?  The `view` issue does not come into play since the pattern matching with Option construct is not even valid if you are dealing with `anOption.view`.